dependencies.yaml

```diff
@@ -0,0 +1,157 @@
+    output: none
+    includes:
+      - cudatoolkit
+      - py_version
+      - test_python
+  checks:
+    output: none
+    includes:
+      - build
+      - develop
+channels:
+  - rapidsai
+  - rapidsai-nightly
+  - conda-forge
+  - nvidia
+dependencies:
+  build_cpp:
+    common:
+      - output_types: conda
+        packages:
+          - &cmake_ver cmake>=3.23.1,!=3.25.0
+          - c-compiler
+          - cxx-compiler
+          - gdal>3.5.0,<3.6.0
+          - libcudf=22.12.*
+          - librmm=22.12.*
+          - ninja
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+            packages:
+              - &gcc_amd64 gcc_linux-64=9.*
+              - &sysroot_amd64 sysroot_linux-64==2.17
+          - matrix:
+              arch: aarch64
+            packages:
+              - &gcc_aarch64 gcc_linux-aarch64=9.*
+              - &sysroot_aarch64 sysroot_linux-aarch64==2.17
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+              cuda: "11.5"
+            packages:
+              - &nvcc_amd64_115 nvcc_linux-64=11.5
+          - matrix:
+              arch: aarch64
+              cuda: "11.5"
+            packages:
+              - &nvcc_aarch64_115 nvcc_linux-aarch64=11.5
+  build_python:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - *cmake_ver
+          - cudf=22.12.*
+          - cython>=0.29,<0.30
+          - scikit-build>=0.13.1
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              arch: x86_64
+            packages:
+              - *gcc_amd64
+              - *sysroot_amd64
+          - matrix:
+              arch: aarch64
+            packages:
+              - *gcc_aarch64
+              - *sysroot_aarch64
+  cudatoolkit:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              cuda: "11.2"
+            packages:
+              - cudatoolkit=11.2
+          - matrix:
+              cuda: "11.4"
+            packages:
+              - cudatoolkit=11.4
+          - matrix:
+              cuda: "11.5"
+            packages:
+              - cudatoolkit=11.5
+  develop:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - pre-commit
+  doc:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - pydata-sphinx-theme
+          - myst-parser
+  py_version:
+    specific:
+      - output_types: conda
+        matrices:
+          - matrix:
+              py: "3.8"
+            packages:
+              - python=3.8
+          - matrix:
+              py: "3.9"
+            packages:
+              - python=3.9
+          - matrix:
+            packages:
+              - python>=3.8,<3.10
+  run_python:
+    common:
+      - output_types: [conda, requirements]
+        packages:
+          - cudf=22.12.*
```

Comment:
@bdice, I see we have `cudf=22.12.*` twice in this file. any objections to using YAML anchors to keep it DRY?

Reply:
It doesn't really matter since these are handled by `ci/release/update-version.sh`. No objections - either way is fine with me.

Reply:
Updated in 56151d16831d16ed49d2e8e637e387a796986384.